snuba/web/rpc/common/eap_execute.py

```diff
@@ -0,0 +1,51 @@
+from concurrent.futures import Future, ThreadPoolExecutor
+from typing import Any, Mapping, Optional
+
+from snuba import settings
+from snuba.clickhouse.formatter.query import format_query
+from snuba.datasets.plans.translator.query import identity_translate
+from snuba.datasets.schemas.tables import TableSource
+from snuba.datasets.storages.factory import get_storage
+from snuba.datasets.storages.storage_key import StorageKey
+from snuba.query import Query
+from snuba.query.data_source.simple import Table
+from snuba.reader import Result as ClickhouseQueryResult
+
+
+def eap_execute_query(
+    query: Query,
+    clickhouse_settings: Optional[Mapping[str, Any]] = None,
+    storage_name: str = "eap_spans",
+) -> ClickhouseQueryResult:
+    storage_key = StorageKey(storage_name)
+    storage = get_storage(storage_key)
+    data_source = storage.get_schema().get_data_source()
+    assert isinstance(data_source, TableSource)
+    cluster = storage.get_cluster()
+    clickhouse_query = identity_translate(query)
+    clickhouse_query.set_from_clause(
+        Table(
+            table_name=data_source.get_table_name(),
+            schema=data_source.get_columns(),
+            storage_key=storage_key,
+        )
+    )
+    formatted_query = format_query(clickhouse_query)
+    reader = cluster.get_reader()
+    return reader.execute(formatted_query, clickhouse_settings, robust=True)
```

Comment:
this should be wrapped in the allocation policies. You can see an example of it here:

https://github.com/getsentry/snuba/blob/master/snuba/web/db_query.py#L600